archetypes/Exchange/index.tsx

```diff
@@ -31,10 +31,16 @@ const TRADE_OPTIONS = [
     },
 ];
 
+type CommitGasTypes = {
+    mint?: 'string';
+    burn?: 'string';
+    flip?: 'string';
+};
```

Comment:
we can make this type Partial<Record<CommitActionEnum, string>>, and pass the commit gas fee in with 

gasFee = {commitGasFees[commitAction]}

Reply:
Not sure I implemented correctly but I kept getting TS errors when trying to define the state type like above.

Passing prop was a bit of struggle.. ended up having to do this:
gasFee={commitGasFees[CommitActionEnum[commitAction] as keyof typeof CommitActionEnum]}